### PR TITLE
Identity confirmation task present on all claims

### PR DESCRIPTION
**Pull request by tekin**
This makes the "Identity confirmation" task present on all claims, and not just those that do not have GOV.UK Verified identity information.

The aim of this change is to make the implicit explicit. To now, service operators have performed the identity confirmation as part of their qualification checks. This change separates out that implicit confirmation into its own separate task. This means we can more accurately reflect why a claim may not be eligible, and also makes it less likely that the identity confirmation isn't carried out properly.

## Data migration

This PR includes a data migration that will back-fill an identity confirmation task for all those claims with a "passed" qualification check:

```
rails runner db/data/20200403093356_backfill_identity_confirmation_tasks.rb
```

## Follow-up work

This PR also simplifies the automated qualification check so that it now _only_ checks the qualifications, and does not check the identity information. A follow up PR will update this automation so that it separately performs an identity confirmation check and creates a task if the data matches that on the claim.

### CODE REVIEW

File: db/data/20200403093356_backfill_identity_confirmation_tasks.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Run this with:
+#
+#   rails runner db/data/20200403093356_backfill_identity_confirmation_tasks.rb
+#
+# Back-fills identity_confirmation tasks for all those claims that have a successfully qualification check. To now,
+# identity confirmation was implied in the qualification check, but we are now recording the identity confirmation
+# explicitly as its own task.
+claims_with_a_passed_qualificaiton_task = Claim.joins(:tasks).merge(Task.where(name: "qualifications", passed: true))
```
[the remark this answers is not in the document]
Comment (robbiepaul) [reply db/data/20200403093356_backfill_identity_confirmation_tasks.rb:8]: It's probably a bit of an edge case, but do we want to exclude claims that didn't go through GOV.UK Verify from this backfill?

[the remark this answers is not in the document]
Comment (tekin) [reply db/data/20200403093356_backfill_identity_confirmation_tasks.rb:8]: No, that’s a really good point, we definitely don’t want to backfill verify-skipped claims!

[the remark this answers is not in the document]
Comment (tekin) [reply db/data/20200403093356_backfill_identity_confirmation_tasks.rb:8]: No, that’s a really good point, we definitely don’t want to backfill verify-skipped claims!

[the remark this answers is not in the document]
Comment (tekin) [reply db/data/20200403093356_backfill_identity_confirmation_tasks.rb:8]: I've updated the data migration to skip non-verified claims.

File: db/data/20200403093356_backfill_identity_confirmation_tasks.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+# explicitly as its own task.
+claims_with_a_passed_qualification_task = Claim.joins(:tasks).merge(Task.where(name: "qualifications", passed: true))
+
+puts "#{claims_with_a_passed_qualification_task.count} claims that have a PASSED qualification check task"
+puts "Back-filling identity confirmation tasks for these claims..."
+
+claims_with_a_passed_qualification_task.each do |claim|
+  if claim.tasks.detect { |task| task.name == "identity_confirmation" }
+    puts "Claim #{claim.id}: identity_confirmation task already complete; skipping"
+  elsif !claim.identity_verified?
```
Comment (robbiepaul) [inline db/data/20200403093356_backfill_identity_confirmation_tasks.rb:16]: 👍

**Pull request merged**